Sources/GRPC/Interceptor/ClientInterceptorPipeline.swift

```diff
@@ -79,16 +79,16 @@ internal final class ClientInterceptorPipeline<Request, Response> {
   internal let _errorDelegate: ClientErrorDelegate?
 
   @usableFromInline
-  internal let _onError: (Error) -> Void
+  internal var _onError: ((Error) -> Void)?
 
   @usableFromInline
-  internal let _onCancel: (EventLoopPromise<Void>?) -> Void
+  internal var _onCancel: ((EventLoopPromise<Void>?) -> Void)?
 
   @usableFromInline
-  internal let _onRequestPart: (GRPCClientRequestPart<Request>, EventLoopPromise<Void>?) -> Void
+  internal var _onRequestPart: ((GRPCClientRequestPart<Request>, EventLoopPromise<Void>?) -> Void)?
 
   @usableFromInline
-  internal let _onResponsePart: (GRPCClientResponsePart<Response>) -> Void
+  internal var _onResponsePart: ((GRPCClientResponsePart<Response>) -> Void)?
 
```

Comment:
If those are now optional, shouldn't we change the initializer?

Reply:
Since, we nil them ourselves later, maybe we can make them:
```
internal private(set) var
```

This would communicate the intent a little better.

Reply:
> If those are now optional, shouldn't we change the initializer?

We only want to `nil` them out when we close.

> Since, we nil them ourselves later, maybe we can make them:
> 
> ```
> internal private(set) var
> ```

Sadly not as they are `@usableFromInline` (the leading `_` indicates they'd be `private` otherwise)

Reply:
I take that back; to my surprise `internal private(set)` works with `@usableFromInline`!